src/GovUk.Education.ExploreEducationStatistics.Data.Model/Repository/FootnoteRepository.cs

```diff
@@ -19,40 +18,134 @@ public FootnoteRepository(StatisticsDbContext context)
+            // Filter out any of these which don't belong to the same subject
+            var filterItemsQueryable = _context.FilterItem
+                .Where(filterItem => filterItemIds.Contains(filterItem.Id)
+                                     && filterItem.FilterGroup.Filter.SubjectId == subjectId);
+
+            var filterItems = await filterItemsQueryable
+                .Select(filterItem => filterItem.Id)
+                .ToListAsync();
+
+            var filterGroups = await filterItemsQueryable
+                .Select(filterItem => filterItem.FilterGroupId)
+                .Distinct()
+                .ToListAsync();
+
+            var filters = await filterItemsQueryable
+                .Select(filterItem => filterItem.FilterGroup.FilterId)
+                .Distinct()
+                .ToListAsync();
+
+            var indicators = await _context.Indicator
+                .Where(indicator => indicatorIds.Contains(indicator.Id)
+                                    && indicator.IndicatorGroup.SubjectId == subjectId)
+                .Select(indicator => indicator.Id)
+                .ToListAsync();
 
-            return _context
+            // A footnote must be linked to the release
+            var footnotesQueryable = _context
                 .Footnote
-                .FromSqlRaw(
-                    "EXEC dbo.FilteredFootnotes " +
-                    "@releaseId," +
-                    "@subjectId," +
-                    "@indicatorList," +
-                    "@filterItemList",
-                    releaseIdParam,
-                    subjectIdParam,
-                    indicatorListParam,
-                    filterItemListParam)
-                .AsNoTracking();
+                .Where(footnote => footnote.Releases.Any(releaseFootnote => releaseFootnote.ReleaseId == releaseId));
+
+            // It must also be linked to any of the filter items, filter groups, filters, indicators, or the subject
+            footnotesQueryable = footnotesQueryable
+                .Where(footnote => !footnote.FilterItems.Any() ||
+                                   footnote.FilterItems.Any(filterItemFootnote =>
+                                       filterItems.Contains(filterItemFootnote.FilterItemId)))
+                .Where(footnote => !footnote.FilterGroups.Any() ||
+                                   footnote.FilterGroups.Any(filterGroupFootnote =>
+                                       filterGroups.Contains(filterGroupFootnote.FilterGroupId)))
+                .Where(footnote => !footnote.Filters.Any() ||
+                                   footnote.Filters.Any(filterFootnote =>
+                                       filters.Contains(filterFootnote.FilterId)))
+                .Where(footnote => !footnote.Indicators.Any() ||
+                                   footnote.Indicators.Any(indicatorFootnote => 
+                                       indicators.Contains(indicatorFootnote.IndicatorId)))
```

Comment:
It's possible that either `filterItems` or `indicators` is empty, meaning it's possible to construct a `Where` against an empty list. 

I can't remember if EF protects against this. If it doesn't, we end up SQL like `WHERE IN ()` which always throws an SQL exception. 

Would suggest better protecting against this if EF doesn't automatically handle potentially empty lists.

Reply:
I'm going to unit test and manually test this method to try and trip it up, especially after other changes have been made following comments on this PR. Will report back here when unit tests are sorted, thanks.

Reply:
Added lots of unit tests which did show up a problem here, not with empty collections and a `WHERE IN ()` but with all the logic between the different `.Where` clauses.

I've simplified the logic now which has made the unit tests happy!

We still end up with lots of conditions like this

```
                footnote.FilterItems.Any(filterItemFootnote =>
                    filterItemIdsSanitised.Contains(filterItemFootnote.FilterItemId)) ||
```

We could prefix these with a check to make sure the lists aren't empty, e.g.

```
                filterItemIdsSanitised.Any() && footnote.FilterItems.Any(filterItemFootnote =>
                    filterItemIdsSanitised.Contains(filterItemFootnote.FilterItemId)) ||
```

but that doesn't seem to be necessary as EF is automatically protecting against this.